internal/executer/executer.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executer
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// Executer is an interface that simulates an executable binary.
+type Executer interface {
+	Execute(ctx context.Context, input io.Reader, action string) ([]byte, error)
+}
+
+// executable implements the Executer interface.
+type executable struct {
+	name string
+}
+
+// NewExecuter returns a new Executer instance.
+func NewExecuter(name string) Executer {
+	return &executable{
```

Comment:
Do weed need to check if the executable can be found? Like using [`exec.LookPath`](https://pkg.go.dev/os/exec#LookPath)?

Reply:
I think we need more discussion on that? This is worth a future "feat" PR. I'll leave it as it is for this PR.

Reply:
It is a good idea and is necessary for detecting `desktop.exe`. However, we can leave it in the next PR.